src/site/shared/tools/WiringTool.ts

```diff
@@ -0,0 +1,109 @@
+import {Vector} from "Vector";
+
+import {CircleContains} from "math/MathUtils";
+
+import {MinDist} from "core/utils/Reducers";
+
+import {Circuit, Port} from "core/public";
+
+import {CircuitDesigner}                       from "shared/circuitdesigner/CircuitDesigner";
+import {LEFT_MOUSE_BUTTON, RIGHT_MOUSE_BUTTON} from "shared/utils/input/Constants";
+import {InputManagerEvent}                     from "shared/utils/input/InputManagerEvent";
+import {Tool}                                  from "./Tool";
+
+
+// The distance away from the port in selection for wiring.
+export const WIRING_PORT_SELECT_RADIUS = 0.34;
+
+export class WiringTool implements Tool {
```

Comment:
Is this all a direct port of the previous impl?

Reply:
Mostly yes, cleaned up parts where I thought I could -- please feel free to comment on its implementation anyways since it's good to talk about

Reply:
Ok, yea, I'm not super tuned-in to the tool state flow, but `findPort` makes sense at least